tools/pythonpkg/src/pyresult.cpp

```diff
@@ -217,8 +217,15 @@ py::dict DuckDBPyResult::FetchNumpyInternal(bool stream, idx_t vectors_per_chunk
 
 	// now that we have materialized the result in contiguous arrays, construct the actual NumPy arrays
 	py::dict res;
+	unordered_map<string, idx_t> names;
 	for (idx_t col_idx = 0; col_idx < result->types.size(); col_idx++) {
-		res[result->names[col_idx].c_str()] = conversion.ToArray(col_idx);
+		if (names[result->names[col_idx]]++ == 0) {
+			res[result->names[col_idx].c_str()] = conversion.ToArray(col_idx);
+		} else {
+			auto name = result->names[col_idx] + "_" + to_string(names[result->names[col_idx]]);
```

Comment:
This still seems incorrect to me, as the code here assumes "name_1" does not exist. What if the columns are in order `id`, `id_1`, `id`? Then the second `id` will overwrite the previous `id_1`. Perhaps re-use the code from [here](https://github.com/duckdb/duckdb/blob/5190fbcd7ad4347b809a37db5cf8fff6c5ae8d23/src/planner/bind_context.cpp#L335)?

Reply:
I think you are partially correct.
we insert id and Id_1 without conflict
so our ordered_map is
{id,1}{id_1,1}

the second one would increment 
{id,2} and generate id_2 correctly.

but I guess id id_1 Id_2 id would generate a problem, so I'm missing a while loop for checking that.